### ISSUE TITLE
NS2001: False positive for class with protected internal constructor

### ISSUE THREAD

**Issue by reduckted** [nsubstitute/NSubstitute.Analyzers#219]
When creating a substitute for a class with a `protected internal` constructor, the diagnostic NS2001 is produced, even though the substitute can be created.

This seems to be a recurrence of #144.

```cs
public class Program {
    public static void Main() {
        Tester target = Substitute.For<Tester>(42);
        //              ~~~~~~~~~~~~~~~~~~~~~~~~~~ Could not find accessible constructor.

        Console.WriteLine(target.GetValue());
        // Outputs 42.
    }
}

public class Tester {
    private readonly int _value;

    protected internal Tester(int value) {
        _value = value;
    }

    public int GetValue() {
        return _value;
    }
}
```
